pyomo/solvers/tests/checks/test_GAMS.py

```diff
@@ -162,5 +165,146 @@ def test_keepfiles_gms(self):
 
             shutil.rmtree(tmpdir)
 
+class GAMSLogfileTestBase(unittest.TestCase):
+    def setUp(self):
+        """Set up model and temporary directory."""
+        m = ConcreteModel()
+        m.x = Var()
+        m.c = Constraint(expr= m.x >= 10)
+        m.o = Objective(expr= m.x)
+        self.m = m
+        self.tmpdir = mkdtemp()
+        self.logfile = os.path.join(self.tmpdir, 'logfile.log')
+
+    def tearDown(self):
+        """Clean up temporary directory after tests are over."""
+        shutil.rmtree(self.tmpdir)
+
+    def _check_logfile(self, exists=True, expected=None):
+        """Check for logfiles existence and contents.
+
+        exists=True:
+            Whether to check if the logfile exists or doesn't exist.
+        expected=None:
+            Optionally check that the logfiles contents is equal to this value.
+
+        """
+        if not exists:
+            self.assertFalse(os.path.exists(self.logfile))
+            return
+
+        self.assertTrue(os.path.exists(self.logfile))
+        with open(self.logfile) as f:
+            logfile_contents = f.read()
+        self.assertTrue(logfile_contents)
```

Comment:
In general, I do not like `assertTrue` / `assertFalse` for testing for non-empty / empty strings.  A much stronger test is to decide on a characteristic string that should be there ("GAMS" is probably a good guess), or to `assertEqual(logfile_contents, "")`.